packages/mdx/test/compile.js

```diff
@@ -548,6 +548,56 @@ test('compile', async () => {
   )
   console.log('\nnote: the preceding warning is expected!\n')
 
+  const developmentSourceNode = (
+    await run(
+      compileSync(
+        {value: '<div />', path: 'path/to/file.js'},
+        {development: true}
+      ).value
+    )
+  )({})
+  assert.equal(
+    // @ts-expect-error React attaches source information on this property,
+    // but it’s private and untyped.
+    developmentSourceNode._source,
+    {fileName: 'path/to/file.js', lineNumber: 1, columnNumber: 1},
+    'should expose source information in the automatic jsx dev runtime'
+  )
```

Comment:
The test above uses an undocumented private property of React elements. The test asserts the generated source code. None of these tests is ideal, but I’m not sure which is worse. I suggest to keep only one of them though.